arbnode/dataposter/data_poster.go

```diff
@@ -291,20 +322,19 @@ func (p *DataPoster[Meta]) saveTx(ctx context.Context, prevTx *queuedTransaction
 
 func (p *DataPoster[Meta]) sendTx(ctx context.Context, prevTx *queuedTransaction[Meta], newTx *queuedTransaction[Meta]) error {
 	if prevTx != newTx {
-		err := p.saveTx(ctx, prevTx, newTx)
-		if err != nil {
+		if err := p.saveTx(ctx, prevTx, newTx); err != nil {
 			return err
 		}
 	}
-	err := p.client.SendTransaction(ctx, newTx.FullTx)
-	if err != nil {
-		if strings.Contains(err.Error(), "already known") || strings.Contains(err.Error(), "nonce too low") {
-			log.Info("DataPoster transaction already known", "err", err, "nonce", newTx.FullTx.Nonce(), "hash", newTx.FullTx.Hash())
-			err = nil
-		} else {
+	if err := p.client.SendTransaction(ctx, newTx.FullTx); err != nil {
+		// TODO: comparing error strings to distinguish errors is a bad practice
+		// in Go. We should instead use Error structure.
+		// https://google.github.io/styleguide/go/best-practices#error-structure
+		if !strings.Contains(err.Error(), "already known") && !strings.Contains(err.Error(), "nonce too low") {
```

Comment:
Deleted the todo.

I assume you mean jsonrpc doesn't support propagating errors properly so that you can check it with errors.Is since geth does seem to define those errors and wrap when returning (e.g. https://github.com/ethereum/go-ethereum/blob/master/core/error.go#L48, https://github.com/ethereum/go-ethereum/blob/2274a03e339f213361453590b54917bbfd0a0c31/core/txpool/validation.go#L192). 

If that's the case, I guess we can't do much. But at least we can use e.g. core.ErrNonceTooLow.Error() instead of hard-coding the string.